tests/lib/config/flat-config-schema.js

```diff
@@ -100,22 +100,28 @@ describe("merge", () => {
         assert.notStrictEqual(result, second);
     });
 
-    it("overwrites a value in the first object with null in the second one", () => {
+    it("throws an error if a value in a property is overriden with null", () => {
```

Comment:
Again this is the current behavior, but this is happening because of the call to `Object.keys(second)` on this line:

https://github.com/eslint/eslint/blob/5de9637fc925729a83d5a5e9e868a41792a184e3/lib/config/flat-config-schema.js#L91

The problem is that `Object.keys` throws an error if called with an argument of `null` or `undefined`. Here, the argument (`second`) cannot be `undefined`, because it is assigned a default value in the declaration, but it can be `null`.

If we only want to fix this test case, and the previous case where a string is being destructured into an object, then we could make sure that `second` is an object before calling `deepMerge` (and if it is a primitive, replace it with an empty object):

```diff
-        const secondValue = second[key];
+        let secondValue = second[key];
 
         if (isNonNullObject(firstValue)) {
+            if (!isNonNullObject(secondValue)) {
+                secondValue = EMPTY_OBJECT;
+            }
```

This would avoid the `TypeError` when the second value is `null`, and treat `"qux"` as an empty object, rather than using its characters as properties, and it would keep the current behavior unchanged in all the other test cases.

Reply:
As per https://github.com/eslint/eslint/issues/17820 - currently the ecosystem relies on the override behaviour - at least for `null`.

So crashing out here will impact migration efforts and will mean that we have to redesign parser options to account for this change.

---

For reference - typescript eslint handles `parserOptions.project = null` to mean "not set" (I.e. Type aware linting is turned off).

The current behaviour is that a user can turn on type-aware linting in their base config (eg `parserOptions.project = ["some TS project path"]`) then pick specific parts of their codebase for which to disable type-aware linting using eslintrc overrides and setting `parserOptions.project = null`.

For eslintrc merging - the `null` clobbers the value - enabling this behaviour.

There's no other way to clobber or erase the value in an overrides because an empty array just merges in and is thus a no-op.

As I said - removing this behaviour would set the ecosystem back unless an alternative is provided. Being able to "clear" a property in an override is quite useful.

Reply:
@fasttime ah okay, thanks for explaining. 

@bradzacher for the purposes of this PR, we don't want to change the default behavior. We'll discuss on your issue.